spyrmsd/graph.py

```diff
@@ -1,42 +1,97 @@
+from spyrmsd import constants
+
+import numpy as np
+import os
+
+available_backends = []
+
 try:
     from spyrmsd.graphs.gt import (
-        cycle,
-        graph_from_adjacency_matrix,
-        lattice,
-        match_graphs,
-        num_edges,
-        num_vertices,
-        vertex_property,
+        cycle as gt_cycle,
+        graph_from_adjacency_matrix as gt_graph_from_adjacency_matrix,
+        lattice as gt_lattice,
+        match_graphs as gt_match_graphs,
+        num_edges as gt_num_edges,
+        num_vertices as gt_num_vertices,
+        vertex_property as gt_vertex_property,
     )
-
+    available_backends.append("graph-tool")
 except ImportError:
-    try:
-        from spyrmsd.graphs.nx import (
-            cycle,
-            graph_from_adjacency_matrix,
-            lattice,
-            match_graphs,
-            num_edges,
-            num_vertices,
-            vertex_property,
-        )
-    except ImportError:
-        raise ImportError("graph_tool or NetworkX libraries not found.")
-
-__all__ = [
-    "graph_from_adjacency_matrix",
-    "match_graphs",
-    "vertex_property",
-    "num_vertices",
-    "num_edges",
-    "lattice",
-    "cycle",
-    "adjacency_matrix_from_atomic_coordinates",
-]
-
-import numpy as np
-
-from spyrmsd import constants
+    pass
+    
+try:
+   from spyrmsd.graphs.nx import (
+        cycle as nx_cycle,
+        graph_from_adjacency_matrix as nx_graph_from_adjacency_matrix,
+        lattice as nx_lattice,
+        match_graphs as nx_match_graphs,
+        num_edges as nx_num_edges,
+        num_vertices as nx_num_vertices,
+        vertex_property as nx_vertex_property,
+    )
+   available_backends.append("networkx")
+except ImportError:
+   pass
+        
+def get_available_backends():
+    return available_backends 
+
+def set_backend(backend):
+
+    ## Get the current backend
+    current_backend = str(os.environ.get("SPYRMSD_BACKEND"))
+  
+    ## Check if the backend is valid
+    if backend.lower() in ["graph_tool", "graphtool", "graph-tool", "graph tool", "gt"]:
+        backend = "graph-tool"
+    elif backend.lower() in ["networkx", "nx"]:
+        backend = "networkx"
+    else:
+        raise ValueError("This backend is not recognized or supported")
+
+    ## Check if we the backend is installed
+    if not backend in available_backends:
+        raise ImportError(f"The {backend} backend doesn't seem to be installed")
+
+    ## Check if we actually need to switch backends
+    if backend == current_backend:
+        print(f"The backend is already {backend}")
+        return
+        
+    global cycle, graph_from_adjacency_matrix, lattice, match_graphs, num_edges, num_vertices, vertex_property
```

Comment:
The `global` keyboard, which I never used in 10+ years of programming in Python, makes me think if there is an even cleaner solution (but possibly more complex)? 

However, if we get the API right (`set_backend`, `get_backend`, ...) we can always change how it is done later. This is why having everything very local and transparent to the user was important to me.

Reply:
I hadn't used the global keyword before either, but seems to work quite well in this case..
I'm open for any suggestions, but I can't immediately think of another/cleaner way (although as you pointed out, it might exist!)